File: src/main/java/exnihilocreatio/registries/types/Compostable.java
package exnihilocreatio.registries.types;

import exnihilocreatio.texturing.Color;
import exnihilocreatio.util.BlockInfo;
import lombok.AllArgsConstructor;
import lombok.Getter;

import javax.annotation.Nonnull;

@AllArgsConstructor
public class Compostable {

    public static final Compostable EMPTY = new Compostable(0f, new Color(0), BlockInfo.EMPTY);

    @Getter
    private float value;
    @Getter
    private Color color;
    @Getter
    @Nonnull
    private BlockInfo compostBlock;
}

File: src/main/java/exnihilocreatio/registries/types/Compostable.kt
package exnihilocreatio.registries.types

import exnihilocreatio.texturing.Color
import exnihilocreatio.util.BlockInfo

data class Compostable(
        val value: Float = 0.toFloat(),
        val color: Color? = null,
        val compostBlock: BlockInfo) {

    companion object {
        val EMPTY = Compostable(0f, Color(0), BlockInfo.EMPTY)
    }
}
